frontend/public/components/modals/delete-modal.tsx

```diff
@@ -140,6 +141,8 @@ export const deleteModal = createModalLauncher(DeleteModal);
 export type DeleteModalProps = {
   kind: K8sKind;
   resource: any;
+  cleanupResource?: boolean;
+  callback?: () => void;
```

Comment:
Callback sounds too generic, `onDelete` would be more descriptive
```suggestion
  onDelete?: () => void;
```